### ISSUE TITLE
[Help Wanted] [MM-10286] Make ConnMaxLifetime configurable

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#8663]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-10286)

We currently expose SqlSettings.MaxIdleConns, SqlSettings.MaxOpenConns and SqlSettings.QueryTimeout, but hard-code:
```
const (
	MAX_DB_CONN_LIFETIME = 60
)

db.SetConnMaxLifetime(time.Duration(MAX_DB_CONN_LIFETIME) * time.Minute)
```

There are situations where it is useful to allow the admin to configure this timeout.

**Comment by jkurian** [comment]
I would like to look into this!

**Comment by jkurian** [comment]
I will be looking into this!

**Comment by jasonblais** [comment]
Thanks @jkurian! :tada

**Comment by jasonblais** [comment]
Thanks @jkurian! This will be available in v5.1. Closing

**Comment by jkurian** [comment]
@jasonblais My pleasure :)
